src/hle.c

```diff
@@ -109,12 +109,35 @@ void hle_init(struct hle_t* hle,
 
 void hle_execute(struct hle_t* hle)
 {
-    if (is_task(hle)) {
-        if (!try_fast_task_dispatching(hle))
-            normal_task_dispatching(hle);
-    } else {
-        non_task_dispatching(hle);
+    uint32_t uc_start = *dmem_u32(hle, TASK_UCODE);
+    uint32_t uc_dstart = *dmem_u32(hle, TASK_UCODE_DATA);
+    uint32_t uc_dsize = *dmem_u32(hle, TASK_UCODE_DATA_SIZE);
+
+    bool match = false;
+    struct ucode_info_t *info = &tasks.info[tasks.count-1];
```

Comment:
> Undefined Behavior when tasks.count == 0

This is covered by the ```for (int i = 0; i < tasks.count; i++)```
You won't enter the loop when tasks.count == 0 so match = false

```
    if (!match)
    {
        info = &tasks.info[tasks.count];
```

I'm iterating backward because the first ucode in list is usually used for boot only, so no need to test it everytime. Otherwise it won't matter since it's constantly swapping between video and audio ucodes like you said.

Reply:
UB us for the tasks.info[task.count-1] part

Reply:
Doesn't matter since the buggy address won't ever be used

Reply:
Don't make me add a if for this shit ;-)